fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteSimpleAggToHDFSScanRule.java

```diff
@@ -0,0 +1,207 @@
+
+        if (scanOperator instanceof LogicalHiveScanOperator) {
+            newMetaScan = new LogicalHiveScanOperator(scanOperator.getTable(),
+                    newScanColumnRefs, newScanColumnMeta, scanOperator.getLimit(), scanOperator.getPredicate());
+        } else if (scanOperator instanceof LogicalIcebergScanOperator) {
+            newMetaScan = new LogicalIcebergScanOperator(scanOperator.getTable(),
+                    newScanColumnRefs, newScanColumnMeta, scanOperator.getLimit(), scanOperator.getPredicate());
+        } else {
+            throw new IllegalStateException("Unknown scan operator: " + scanOperator);
+        }
+        try {
+            newMetaScan.setScanOperatorPredicates(scanOperator.getScanOperatorPredicates());
+        } catch (AnalysisException e) {
+            throw new IllegalStateException("Set scan operator predicates", e);
+        }
+        LogicalAggregationOperator newAggOperator = new LogicalAggregationOperator(aggregationOperator.getType(),
+                aggregationOperator.getGroupingKeys(), newAggCalls);
+
+        newAggOperator.setProjection(aggregationOperator.getProjection());
+        OptExpression optExpression = OptExpression.create(newAggOperator);
+        optExpression.getInputs().add(OptExpression.create(newMetaScan));
+        return optExpression;
+    }
+
+    @Override
+    public boolean check(final OptExpression input, OptimizerContext context) {
+        if (!context.getSessionVariable().isEnableRewriteSimpleAggToHdfsScan()) {
+            return false;
+        }
+        LogicalAggregationOperator aggregationOperator = (LogicalAggregationOperator) input.getOp();
+        LogicalScanOperator scanOperator = (LogicalScanOperator) input.getInputs().get(0).getInputs().get(0).getOp();
+
+        // no limit
+        if (scanOperator.getLimit() != -1) {
+            return false;
+        }
+
+        // filter only involved with partition keys.
+        if (scanOperator.getPredicate() != null) {
+            if (!scanOperator.getPartitionColumns()
+                    .containsAll(scanOperator.getPredicate().getColumnRefs().stream().map(x -> x.getName()).collect(
+                            Collectors.toList()))) {
+                return false;
+            }
+        }
+
+        // all group by keys are partition keys.
+        List<ColumnRefOperator> groupingKeys = aggregationOperator.getGroupingKeys();
+        if (!scanOperator.getPartitionColumns()
+                .containsAll(groupingKeys.stream().map(x -> x.getName()).collect(Collectors.toList()))) {
+            return false;
+        }
+        // TODO(yanz): not quite sure why this limitation !
+        if (aggregationOperator.getPredicate() != null) {
+            return false;
+        }
+
+        if (scanOperatorType == OperatorType.LOGICAL_ICEBERG_SCAN) {
+            IcebergTable icebergTable = (IcebergTable) scanOperator.getTable();
+            if (!icebergTable.isAllPartitionColumnsAlwaysIdentity()) {
```

Comment:
Check if it is a partition table

Reply:
I think so, if there is no partition, we don't need to check partitiion column info.